gradle-plugins/compose/src/main/kotlin/org/jetbrains/compose/experimental/internal/checkExperimentalTargets.kt

```diff
@@ -6,31 +6,41 @@
 package org.jetbrains.compose.experimental.internal
 
 import org.gradle.api.Project
+import org.jetbrains.compose.experimental.dsl.ExperimentalExtension
 import org.jetbrains.kotlin.gradle.dsl.KotlinMultiplatformExtension
 import org.jetbrains.kotlin.gradle.plugin.KotlinTarget
 
 private const val SKIKO_ARTIFACT_PREFIX = "org.jetbrains.skiko:skiko"
 
 private class TargetType(
     val id: String,
-    val presets: List<String>
+    val presets: List<String>,
+    /**
+     * @return true if target is experimental
+     */
+    val experimentalCondition: (ExperimentalExtension) -> Boolean
 )
+
 private val TargetType.gradlePropertyName get() = "org.jetbrains.compose.experimental.$id.enabled"
 
 private val EXPERIMENTAL_TARGETS: Set<TargetType> = setOf(
-    TargetType("uikit", presets = listOf("iosSimulatorArm64", "iosArm64", "iosX64")),
-    TargetType("macos", presets = listOf("macosX64", "macosArm64")),
-    TargetType("jscanvas", presets = listOf("jsIr", "js")),
+    TargetType("uikit", presets = listOf("iosSimulatorArm64", "iosArm64", "iosX64")) { true },
+    TargetType("macos", presets = listOf("macosX64", "macosArm64")) { true },
+    TargetType("jscanvas", presets = listOf("jsIr", "js")) { experimentalExtension ->
+        experimentalExtension.web._isApplicationInitialized
```

Comment:
Can JS canvas work if we omit DSL in `build.gradle`?
```
compose.experimental {
    web.application {}
```

Reply:
Theoretically yes. If user add's dependency "org.jetbrains.skiko:skiko-js-wasm-runtime:$version" by hands

Reply:
And unpack skiko wasm binary in correct directory with js resources

Reply:
Also, it is strange that we have skiko in compose runtime JS dependencies. If we have it, we should remove this dependency.

cc @eymar 

Does the project fail in web-template?

Reply:
We never had skiko dependency in runtime. I'll double check tomorrow.

Reply:
Sorry, I have not correct say about stable JS build.
@eymar Yeah, you correct! org.jetbrains.skiko:skiko not exist's in stable js-html projects.

Also if newer projects have dependencies to skiko, it means that js is in experimental mode.
So, previously created PR work's good.
Current PR will be closed.